sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDao.java

```diff
@@ -754,54 +754,139 @@ public List<Case> queryByCriteria(CaseCriteria criteria, long offset, long limit
 	private QueryBuilder<Case, Long> buildQueryBuilder(CaseCriteria criteria) throws SQLException {
 		QueryBuilder<Case, Long> queryBuilder = queryBuilder();
 		QueryBuilder<Person, Long> personQueryBuilder = DatabaseHelper.getPersonDao().queryBuilder();
+		Where<Case, Long> where = queryBuilder.where().eq(AbstractDomainObject.SNAPSHOT, false);
+
+		if (criteria != null) {
+			where.and();
+			if (criteria.getIncludeCasesFromOtherJurisdictions() != null) {
+				createJurisdictionFilter(where, criteria);
+			}
+			createCriteriaFilter(where, criteria);
+			queryBuilder.setWhere(where);
+		}
+
+		queryBuilder = queryBuilder.leftJoin(personQueryBuilder);
+		return queryBuilder;
+	}
+
+	public Where createJurisdictionFilter(Where<Case, Long> where, CaseCriteria caseCriteria) throws SQLException {
+		List<Where> whereUserFilterStatements = new ArrayList<>();
 
-		List<Where<Case, Long>> whereStatements = new ArrayList<>();
-		Where<Case, Long> where = queryBuilder.where();
-		whereStatements.add(where.eq(AbstractDomainObject.SNAPSHOT, false));
+		User currentUser = ConfigProvider.getUser();
+		if (currentUser == null) {
+			return null;
+		}
+
+		final JurisdictionLevel jurisdictionLevel = currentUser.getJurisdictionLevel();
+
+		if (jurisdictionLevel != JurisdictionLevel.NATION) {
+			//whoever created the case or is assigned to it is allowed to access it
+			if (caseCriteria.getIncludeCasesFromOtherJurisdictions().equals(true)) {
+				whereUserFilterStatements.add(
+					where.or(
+						where.eq(Case.REPORTING_USER, currentUser.getId()),
+						where.eq(Case.SURVEILLANCE_OFFICER, currentUser.getId()),
+						where.eq(Case.CASE_OFFICER, currentUser.getId())));
+			}
+
+			switch (jurisdictionLevel) {
+			case REGION:
```

Comment:
Nation and region level are not necessary; users with these jurisdiction levels don't have access to the mobile app